arangosh/Import/quick_hist.h

```diff
@@ -0,0 +1,265 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2017 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Matthew Von-Maszewski
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGODB_IMPORT_QUICK_HIST_H
+#define ARANGODB_IMPORT_QUICK_HIST_H 1
+
+#include <chrono>
+#include <ctime>
+#include <future>
+#include <iomanip>
+#include <mutex>
+#include <numeric>
+#include <vector>
+
+#include "Basics/ConditionLocker.h"
+#include "Basics/ConditionVariable.h"
+#include "Basics/Thread.h"
+#include "Logger/Logger.h"
+
+namespace arangodb {
+namespace import {
+//
+// quickly written histogram class for debugging
+//
+  class QuickHistogram : public arangodb::Thread {
+ private:
+  QuickHistogram(QuickHistogram const&) = delete;
+  QuickHistogram& operator=(QuickHistogram const&) = delete;
+
+ public:
+  QuickHistogram()
+    : Thread("QuickHistogram"), _threadRunning(false)
+  {};
+
+  ~QuickHistogram()
+  {
+    shutdown();
+  }
+
+  void beginShutdown() override {
+    _threadRunning=false;
+    Thread::beginShutdown();
+
+    // wake up the thread that may be waiting in run()
+    CONDITION_LOCKER(guard, _condvar);
+    guard.broadcast();
+  }
+
+
+  void post_latency(std::chrono::microseconds latency) {
+    if (_threadRunning.load()) {
+      CONDITION_LOCKER(guard, _condvar);
+
+      _writingLatencies->push_back(latency);
+    }
+  }
+
+protected:
+  void run() override
+  {
+    _interval_start=std::chrono::steady_clock::now();
+    _measuring_start = _interval_start;
```

Comment:
inconsistent use of whitespace. Line 81 uses whitespace for the assignment, and line 80 does not.
According to our coding conventions the whitespace should be used everywhere and consistently.

Reply:
fixed